src/TestProblem/ELBDM/HaloMerger/Init_TestProb_ELBDM_HaloMerger.cpp

```diff
@@ -634,53 +777,226 @@ void SetParameter()
+                                     index_halo_input );
+                  }
+
+               } // if ( HaloMerger_Halo_InitMode == 1 )
+               else
+                  Aux_Error( ERROR_INFO, "unsupported initialization mode (%s = %d) !!\n",
+                             "HaloMerger_Halo_InitMode", HaloMerger_Halo_InitMode );
+
+            } // for (int index_halo=0; index_halo<HaloMerger_Halo_Num; index_halo++)
+
+            // check whether the input particle cloud overlaps with the input solitons
+            for (int index_soliton=0; index_soliton<HaloMerger_Soliton_Num; index_soliton++)
+            {
+               const int index_soliton_input = index_soliton+1; // index of soliton in the input file
+
+               bool isOverlap = true;
+
+               if ( sqrt( SQR( HaloMerger_ParCloud_CenCoord[index_parcloud][0] - HaloMerger_Soliton_CenCoord[index_soliton][0] )
+                        + SQR( HaloMerger_ParCloud_CenCoord[index_parcloud][1] - HaloMerger_Soliton_CenCoord[index_soliton][1] )
+                        + SQR( HaloMerger_ParCloud_CenCoord[index_parcloud][2] - HaloMerger_Soliton_CenCoord[index_soliton][2] ) )
+                    > ( HaloMerger_ParCloud_DensProf_MaxR[index_parcloud] + 3.0*HaloMerger_Soliton_CoreRadius[index_soliton] ) )
+                  isOverlap = false;
+
+               if ( MPI_Rank == 0 )
+               {
+                  if ( isOverlap )
+                     Aux_Message( stderr, "WARNING : ParCloud_%d (center: [%13.6e, %13.6e, %13.6e], MaxR: %13.6e) overlaps with the Soliton_%d 3r_c-range (center: [%13.6e, %13.6e, %13.6e], r_c: %13.6e) !!\n",
+                                  index_parcloud_input,
+                                  HaloMerger_ParCloud_CenCoord[index_parcloud][0], HaloMerger_ParCloud_CenCoord[index_parcloud][1],
+                                  HaloMerger_ParCloud_CenCoord[index_parcloud][2], HaloMerger_ParCloud_DensProf_MaxR[index_parcloud],
+                                  index_soliton_input,
+                                  HaloMerger_Soliton_CenCoord[index_soliton][0], HaloMerger_Soliton_CenCoord[index_soliton][1],
+                                  HaloMerger_Soliton_CenCoord[index_soliton][2], HaloMerger_Soliton_CoreRadius[index_soliton] );
+               }
+
+            } // for (int index_soliton=0; index_soliton<HaloMerger_Soliton_Num; index_soliton++)
+
+         } // if ( HaloMerger_ParCloud_InitMode == 1 )
+         else
+            Aux_Error( ERROR_INFO, "unsupported initialization mode (%s = %d) !!\n",
+                       "HaloMerger_ParCloud_InitMode", HaloMerger_ParCloud_InitMode );
+
+      } // for (int index_parcloud=0; index_parcloud<HaloMerger_ParCloud_Num; index_parcloud++)
+
+      // check there are particles
+      if ( HaloMerger_ParCloud_NPar_Total <= 0 )
+         Aux_Error( ERROR_INFO, "Total number of particles (sum of HaloMerger_ParCloud_i_NPar) must be >0 for HaloMerger_ParCloud_Num > 0 !!\n" );
```

Comment:
Print out the numbers of `HaloMerger_ParCloud_NPar_Total` and `HaloMerger_ParCloud_Num`